### PR TITLE
add systemd run0 as a become method

### PR DESCRIPTION
**Pull request by konstruktoid**
##### SUMMARY

This PR adds support for `run0` as a become method.
`run0` will be available in [systemd 256](https://github.com/systemd/systemd/releases/tag/v256-rc1), see https://mastodon.social/@pid_eins/112353324518585654  

##### ISSUE TYPE

- New Module/Plugin Pull Request

##### COMPONENT NAME

`community.general.run0`

### CODE REVIEW

File: plugins/become/run0.py [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+              - section: run0_become_plugin
+                key: flags
+            vars:
+              - name: ansible_become_flags
+              - name: ansible_run0_flags
+            env:
+              - name: ANSIBLE_BECOME_FLAGS
+              - name: ANSIBLE_RUN0_FLAGS
+    notes:
+      - This plugin will only work when a polkit rule is in place.
```
Comment (felixfontein) [inline plugins/become/run0.py:66]:
```suggestion
      - This plugin will only work when a polkit rule is in place..
```

File: plugins/become/run0.py [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+              - section: run0_become_plugin
+                key: executable
+            vars:
+              - name: ansible_become_exe
+              - name: ansible_run0_exe
+            env:
+              - name: ANSIBLE_BECOME_EXE
+              - name: ANSIBLE_RUN0_EXE
+        become_flags:
+            description: Options to pass to run0
```
Comment (felixfontein) [inline plugins/become/run0.py:52]:
```suggestion
            description: Options to pass to run0.
```

File: plugins/become/run0.py [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+              - section: run0_become_plugin
+                key: user
+            vars:
+              - name: ansible_become_user
+              - name: ansible_run0_user
+            env:
+              - name: ANSIBLE_BECOME_USER
+              - name: ANSIBLE_RUN0_USER
+        become_exe:
+            description: run0 executable
```
Comment (felixfontein) [inline plugins/become/run0.py:38]:
```suggestion
            description: The C(run0) executable.
```

File: plugins/become/run0.py [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+    name: run0
+    short_description: Systemd's run0
+    description:
+        - This become plugins allows your remote/login user to execute commands as another user via the C(run0) utility.
+    author:
+        - Thomas Sjögren (@konstruktoid)
+    version_added: '8.7.0'
+    options:
+        become_user:
+            description: User you 'become' to execute the task
```
Comment (felixfontein) [inline plugins/become/run0.py:24]:
```suggestion
            description: User you 'become' to execute the task.
```

File: plugins/become/run0.py [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+              - section: privilege_escalation
+                key: become_user
+              - section: run0_become_plugin
+                key: user
+            vars:
+              - name: ansible_become_user
+              - name: ansible_run0_user
+            env:
+              - name: ANSIBLE_BECOME_USER
+              - name: ANSIBLE_RUN0_USER
```
Comment (felixfontein) [inline plugins/become/run0.py:36]: Please also add types to the other options.
```suggestion
              - name: ANSIBLE_RUN0_USER
            type: string
```

File: plugins/become/run0.py [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+              - name: ansible_run0_flags
+            env:
+              - name: ANSIBLE_BECOME_FLAGS
+              - name: ANSIBLE_RUN0_FLAGS
+    notes:
+      - This plugin will only work when a polkit rule is in place.
+"""
+
+EXAMPLES = r"""
+# An example polkit rule that allows the user C(ansible) in the C(wheel) group
```
Comment (felixfontein) [inline plugins/become/run0.py:70]:
```suggestion
# An example polkit rule that allows the user 'ansible' in the 'wheel' group
```

File: plugins/become/run0.py [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+    require_tty = (
+        True  # see https://github.com/ansible-collections/community.general/issues/6932
+    )
+
+    @staticmethod
+    def remove_ansi_codes(line):
+        return ansi_color_codes.sub(b"", line)
+
+    def build_become_command(self, cmd, shell):
+        super(BecomeModule, self).build_become_command(cmd, shell)
```
Comment (felixfontein) [inline plugins/become/run0.py:101]: Why not
```suggestion
        super().build_become_command(cmd, shell)
```
here as well?

**Review by felixfontein — COMMENTED** [review]
Thanks for your contribution! Please note that you need to add tests. (Most other become plugins use unit tests to ensure that the correct command line is built, that's probably a good idea here as well.)

File: plugins/become/run0.py (hunk 1 above)
Comment (konstruktoid) [reply plugins/become/run0.py:66]: is it supposed to be two dots @felixfontein?

Comment (felixfontein) [reply plugins/become/run0.py:66]: Hmm, nope, probably just a sign to me to go to bed soon ;-)

File: plugins/become/run0.py (hunk 7 above)
Comment (konstruktoid) [reply plugins/become/run0.py:104]: i leeched from `machinectl` on that one.

**Comment by konstruktoid** [comment]
unit test will be fixed and extended

**Review by felixfontein — COMMENTED** [review]
Looks good to me! If nobody objects, I'll merge this in ~a week.

**Comment by felixfontein** [comment]
The CI failures are unrelated.

**Comment by felixfontein** [comment]
@konstruktoid thanks for your contribution!

**Pull request merged**